src/ripple/app/consensus/RCLConsensus.cpp

```diff
@@ -843,7 +843,8 @@ RCLConsensus::validate(RCLCxLedger const& ledger, bool proposing)
     v->setTrusted();
     // suppress it if we receive it - FIXME: wrong suppression
     app_.getHashRouter().addSuppression(signingHash);
-    app_.getValidations().addValidation(v, "local");
+    if (! app_.getValidations().addValidation(v, "local"))
```

Comment:
I'm wondering if we can move the `getMaxLedger` check to earlier in this function. It seems like we only need to be checking our own validations, not all received validations.